### ISSUE TITLE
Incorrect err_path

### ISSUE THREAD

**Issue by lukasmacko** [CESNET/libyang#247]
```C
#include <stdio.h>
#include <stdlib.h>
#include <libyang/libyang.h>

int main(int argc, char **argv) {

    struct ly_ctx *ctx = ly_ctx_new(".");
    ly_ctx_load_module(ctx, "example-module", NULL);

    struct ly_set *set = lys_find_xpath(ctx, NULL, "/example-module:container/unknown-augment:unknown", 0);
    printf("Errmsg: %s\nErrpath: %s\n\n", ly_errmsg(), ly_errpath());

    ly_ctx_destroy(ctx, NULL);
}
```

produces output
```
libyang[0]: Unknown module "unknown-augment" relative to the context node "/ietf-yang-library:modules-state".
libyang[0]: Resolving XPath expression "/example-module:container/unknown-augment:unknown" failed. (path: /ietf-yang-library:modules-state)
Errmsg: Resolving XPath expression "/example-module:container/unknown-augment:unknown" failed.
Errpath: /ietf-yang-library:modules-state
```
I would expect err_path to be `/example-module:container`

[example-module.yang](https://github.com/sysrepo/sysrepo/blob/9e78f26ebf46c94ae82f5cbed02a883be842cee0/tests/yang/example-module.yang)

**Comment by michalvasko** [comment]
Hi,
after the recent API change of `lys_find_xpath` it makes sense and I changed it so that it produces the expected error (path is generated from the first current context node), but I think there are other rarer cases when the error will be exactly as it was in this case. The main reason for this is that this function is used to execute XPath on schema, which is kind-of our invention and we tried to stick to the definition and behavior of standard XPath as much as possible, but there were various issues and you found one of them. Also, internally the evaluation was optimized for checking `must` and `when` conditions on schema, in which case I left the behavior as it is meaning the path displayed is the original context node, not the last context node. What I am trying to say is that this function may produce unexpected errors (the result should be always correct and as anticipated), which will technically be fine.

Regards,
Michal
